lib/puma/thread_pool.rb

```diff
@@ -281,8 +281,11 @@ def auto_reap!(timeout=5)
     end
 
     # Tell all threads in the pool to exit and wait for them to finish.
+    # Wait +timeout+ seconds then raise +ForceShutdown+ in remaining threads.
+    # Next, wait an extra +grace+ seconds then force-kill remaining threads.
+    # Finally, wait +kill_grace+ seconds for remaining threads to exit.
     #
-    def shutdown(timeout=-1)
+    def shutdown(timeout=-1, grace: SHUTDOWN_GRACE_TIME, kill_grace: 1)
```

Comment:
Nothing uses these options it appears except the tests so we should probably not even bother to expose them.

Reply:
I noticed this but I'm neutral since this is an internal API.

Reply:
This was only to limit the duration of the added test (which would otherwise take 5 seconds), see 09702d19fed95cb4d6b5cc5a478d2d376db8db3e for an alternate approach that doesn't expose any extra arguments and instead uses a simple `stub_const` patch (added to the test helper).